unit/solvers/refinement/string_constraint_instantiation/instantiate_not_contains.cpp

```diff
@@ -193,28 +193,30 @@ SCENARIO("instantiate_not_contains",
     exprt res=generator.add_axioms_for_function_application(func);
     std::string axioms;
     std::vector<string_not_contains_constraintt> nc_axioms;
-    for(exprt axiom : generator.get_axioms())
+
+    for(string_constraintt sc : generator.get_constraints())
     {
-      simplify(axiom, ns);
-      if(axiom.id()==ID_string_constraint)
-      {
-        string_constraintt sc=to_string_constraint(axiom);
-        axioms+=from_expr(ns, "", sc);
-      }
-      else if(axiom.id()==ID_string_not_contains_constraint)
-      {
-        string_not_contains_constraintt sc=
-          to_string_not_contains_constraint(axiom);
-        axioms+=from_expr(ns, "", sc);
-        generator.witness[sc]=
-          generator.fresh_symbol("w", t.witness_type());
-        nc_axioms.push_back(sc);
-      }
-      else
-        axioms+=from_expr(ns, "", axiom);
+      simplify(sc, ns);
+      axioms += from_expr(ns, "", sc);
+      axioms += "\n\n";
+    }
 
+    for(string_not_contains_constraintt sc :
+        generator.get_not_contains_constraints())
+    {
+      simplify(sc, ns);
+      axioms += from_expr(ns, "", sc);
+      generator.witness[sc] = generator.fresh_symbol("w", t.witness_type());
+      nc_axioms.push_back(sc);
```

Comment:
If you were to break this up into two loops, you could reuse a lambda for the 3 loops and turn this into `std::copy`.

Reply:
What would be the type of this lambda?  The 3 vectors have different types.